src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -80,6 +81,22 @@ public void EndIntervalMeasure(string propertyName)
             _intervalList.Add(new Tuple<string, TimeSpan>(propertyName, _intervalWatch.Elapsed));
         }
 
+        /// <summary> Start interval measure. </summary>
+        /// <param name="stopWatch"> Stopwatch used to measure the time interval. </param>
+        public void StartIntervalMeasure(Stopwatch stopWatch)
+        {
+            stopWatch.Restart();
+        }
```

Comment:
I don't think this is a good API. This class `TelemetryActivity` doesn't "own" the stopwatch. This API doesn't provide the caller any guarantees, it's a trivial wrapper of `stopWatch.Resart`.